tests/test_dataset.py

```diff
@@ -0,0 +1,111 @@
+"""Unit test related to the dataset creation, population and loading
+"""
+
+import json
+import numpy as np
+import os
+
+from deeposlandia.dataset import Dataset, MapillaryDataset, ShapeDataset
+
+def test_dataset_creation():
+    """Create a generic dataset
+    """
+    IMAGE_SIZE = 128
+    d = Dataset(IMAGE_SIZE)
+    assert d.image_size == IMAGE_SIZE
+    assert d.get_nb_labels() == 0
+    assert d.get_nb_images() == 0
+    d.add_label(label_id=0, label_name="test_label", color=[100, 100, 100], is_evaluate=True)
+    assert d.get_nb_labels() == 1
+
+def test_mapillary_dataset_creation():
+    """Create a Mapillary dataset from a configuration file
+    """
+    IMAGE_SIZE = 224
+    config_filename = "tests/data/mapillary/config_aggregate.json"
+    with open(config_filename) as fobj:
+        config = json.load(fobj)
+    NB_LABELS = len(config["labels"])
+    d = MapillaryDataset(IMAGE_SIZE, config_filename)
+    assert d.image_size == IMAGE_SIZE
+    assert d.get_nb_labels(see_all=True) == NB_LABELS
+    assert d.get_nb_images() == 0
+
+def test_mapillary_dataset_population():
```

Comment:
When you test the creation of several image files from a specific dataset, you modify the working directory. See the `git status` after a `pytest tests/test_dataset.py`.

I think, we should have:

* a directory to input data (JSON, some images, etc.)
* a directory to output data which has to be cleaned after each unit test. I think we can add some pytest fixtures to create output dir on the fly, generate files, check some stuff, and remove the output dir.

Reply:
I previously mentioned the [pytest fixtures](https://docs.pytest.org/en/latest/fixture.html). I think it's a proper and cleaner way to handle temporary directories. See the following example https://docs.pytest.org/en/latest/tmpdir.html